src/TTrigger.cpp

```diff
@@ -419,155 +432,158 @@ bool TTrigger::match_perl(char* subject, const QString& toMatch, int regexNumber
         }
     }
 
-END : {
+    END : {
```

Comment:
Uh, oh, we are using `goto`s in here - would it be cleaner to refactor the code here to a method that can be called from where the `goto`s originate, followed by an appropriate `return` from *this* method?

Reply:
Agreed, but that is existing processing code that I'm not really touching too much here.

Reply:
Ah, I've been doing a bit more reading and sometimes, just *sometimes* some "`goto`s the cleanup portion at the end of a function/method" is the best way to **D**on't **R**epeat **Y**ourself...

Reply:
Qt has a special class that can do that now. I don't remember it off the top of my head.

Reply:
[QScopeGuard](https://doc.qt.io/qt-5/qscopeguard.html)

Reply:
Ah, given that that class is a relative new one (it was introduced in Qt **5.12**) you'll be able to use it now, especially once we formally drop supporting Qt 5.12 (LTS) and go to - I guess a minimum of 5.14 (unless we have a fix for the Windows issue for Qt 5.15.2 {+KDE patches?} and can switch to that) - as https://github.com/Mudlet/Mudlet/issues/5623#issuecomment-974564816 points out 5.12 is now EOL...